### PR TITLE
Updated README with API spec and description of application

### PR DESCRIPTION
**Pull request by cgalarza**
An initial draft of the README. Looking for feedback and edits!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,279 @@
...
+## API v1
+
+### Vocabularies
+#### List Vocabularies
+- **Request**
+  `GET /vocabularies`
+
+- **Success Response**
+
+  Sorts vocabularies alphabetically by label. Defaults to `page = 1` and `per_page = 20`.
```
Comment (elohanlon) [inline README.md:42]: We should update the vocabularies response example below to have the terms sorted alphabetically by label.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,279 @@
...
+     "label": "Names"
+  }
+  ```
+
+#### Create Vocabulary
+- **Request**
+
+  `POST /vocabularies`
+
+  `string_key` is a required parameter.
```
Comment (elohanlon) [inline README.md]: `label` is also a required parameter for a new vocabulary.

File: README.md [hunk 3]
```diff
@@ -1,24 +1,279 @@
...
+
+  _Body:_
+    ```json
+    {
+      "string_key": "names",
+      "label": "Names"
+    }
+    ```
+
+#### Update Vocabulary
```
Comment (elohanlon) [inline README.md:112]: Would be good to mention here that the only property of a vocabulary that you're allowed to update is the `label` (and `string_key` should never change).

File: README.md [hunk 4]
```diff
@@ -1,24 +1,279 @@
...
+  `DELETE /vocabularies/:string_key`
+
+- **Success Response**
+
+  _Status:_ 204
+
+  _Body:_ nil
+
+- **Error Response**
+  - If resource does not exists returns `404`
```
Comment (elohanlon) [inline README.md]:
```suggestion
  - If resource does not exist returns `404`
```

File: README.md [hunk 5]
```diff
@@ -1,24 +1,279 @@
...
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+**URI Service** is a standalone Rails 5 application that creates/stores local and temporary terms and caches external URI terms. To start the application's only interface will be a JSON api, though we see expanding the application to include html pages for local terms.
 
-Things you may want to cover:
+URI Service makes some assumptions about how URIs should be organized. A URI is always related to a vocabulary. A vocabulary has many URI terms. Multiple authorities might be represented within a vocabulary. A URI must be unique to a vocabulary, but a URI can appear in multiple vocabularies.
 
-* Ruby version
+In order to use the JSON api, URI Service requires token authentication. URI Service does not provide granular authorization. The application that uses the api should control authorization in anyway it deems necessary.
```
Comment (elohanlon) [inline README.md]:
```suggestion
In order to use the JSON api, URI Service requires token authentication. URI Service does not provide granular authorization. The application that uses the api should control authorization in any way it deems necessary.
```

File: README.md (hunk 2 above)
Comment (cgalarza) [reply README.md]: should this also be required in the model? we had not talked about this requirement

File: README.md [hunk 6]
```diff
@@ -1,24 +1,279 @@
...
+
+
+#### Send back description/instruction
+- **Request**
+
+  `OPTIONS /vocabularies/`
+- **Success Response**
+  TODO
+
+### Vocabularies/Term
```
Comment (elohanlon) [inline README.md]:
```suggestion
### Vocabularies/Terms
```

File: README.md [hunk 7]
```diff
@@ -1,24 +1,279 @@
...
+  Examples:
+  ```
+  GET /vocabularies/:string_key/terms?page=1&per_page=20`
+  GET /vocabularies/:string_key/terms?q=smith&authority=naf&page=1&per_page=20`
+  GET /vocabularies/:string_key/terms?uri=https://creativecommons.org/publicdomain/zero/1.0/
+  GET /vocabularies/:string_key/terms?value=Smith # returns temp term b/c no uri was given
+  ```
+
+  Allowed query fields:
+    - `q`: query uri terms
```
Comment (elohanlon) [inline README.md]:
```suggestion
    - `q`: query term labels using fuzzy matching
```

File: README.md [hunk 8]
```diff
@@ -1,24 +1,279 @@
...
+    - `authority`: facet by authority
+    - `uri`: search for uri
```
Comment (elohanlon) [inline README.md]:
```suggestion
    - `uri`: search by exact uri string
```

File: README.md [hunk 9]
```diff
@@ -1,24 +1,279 @@
...
+    - `value`: search for value with in temp terms
```
Comment (elohanlon) [inline README.md]:
```suggestion
    - `label`: search by exact label string
```

File: README.md [hunk 10]
```diff
@@ -1,24 +1,279 @@
...
+- **Request**
+
+  `GET /vocabularies/:string_key/terms`
+
+  Examples:
+  ```
+  GET /vocabularies/:string_key/terms?page=1&per_page=20`
+  GET /vocabularies/:string_key/terms?q=smith&authority=naf&page=1&per_page=20`
+  GET /vocabularies/:string_key/terms?uri=https://creativecommons.org/publicdomain/zero/1.0/
+  GET /vocabularies/:string_key/terms?value=Smith # returns temp term b/c no uri was given
```
Comment (elohanlon) [inline README.md]:
```suggestion
  GET /vocabularies/:string_key/terms?label=Smith # returns temp term b/c no uri was given
```

File: README.md [hunk 11]
```diff
@@ -1,24 +1,279 @@
...
+  ```
+
+  Allowed query fields:
+    - `q`: query uri terms
+    - `authority`: facet by authority
+    - `uri`: search for uri
+    - `value`: search for value with in temp terms
+
```
Comment (elohanlon) [inline README.md]:
```suggestion
- `starts_with`: return terms that start with the given string

```
Should this be a case insensitive search?

File: README.md [hunk 12]
```diff
@@ -1,24 +1,279 @@
...
+  Other allowed params:
+    - `per_page`: number of terms per page, default: 20
+    - `page`: page to display
+
+- **Success Response**
+
+  Sorting alphabetically by ????
```
Comment (elohanlon) [inline README.md]:
```suggestion
  Sorted primarily by match score (in the case of a q search), secondarily by label alphabetical sort
```

Comment (elohanlon) [reply README.md]: FYI: by "match score" I mean by whatever order solr gives us when it returns results. on the solr side, we'll primarily sort by score, and the second sort comparison will be alphabetical label sort.

File: README.md [hunk 13]
```diff
@@ -1,24 +1,279 @@
...
+    "total records": 55,
+    "terms": [
+      {
+        "pref_label": 'Carla Galarza',
+        "uri":"https://id.library.columbia.edu/term/1111-2222-3333-...",
+        "authority": "local",
+        "custom_field_1": "something"
+      },
+      {
+        "pref_label": Eric O'Hanlon',
```
Comment (elohanlon) [inline README.md]:
```suggestion
        "pref_label": 'Eric O'Hanlon',
```

File: README.md [hunk 14]
```diff
@@ -1,24 +1,279 @@
...
+    ]
+  }
+  ```
+
+#### Create term
+- **Request**
+
+  `POST /vocabularies/:string_key/terms`
+
+  _Required params:_ term_type, label, uri
```
Comment (elohanlon) [inline README.md]:
```suggestion
  _Required params:_ term_type, label, uri (unless term_type = local or temporary)
```

File: README.md [hunk 15]
```diff
@@ -1,24 +1,279 @@
...
+
+  _Optional params:_ Any vocabulary-specific custom fields
```
Comment (elohanlon) [inline README.md]:
```suggestion
  _Optional params:_ authority, and any vocabulary-specific custom fields
```

File: README.md [hunk 16]
```diff
@@ -1,24 +1,279 @@
...
+  _Status:_ 201
+
+  _Body:_ json representation of object
+
+#### Update Term
+- **Request**
+
+  `PATCH /vocabularies/:string_key/terms/:uri`
+
+  URI should be encoded
```
Comment (elohanlon) [inline README.md]:
```suggestion
  URI should be URL-encoded
```

File: README.md [hunk 17]
```diff
@@ -1,24 +1,279 @@
...
+
+- **Error Response**
+  - return 404 (if object cannot be found)
+
+#### Delete term
+- **Request**
+
+  `DELETE  /vocabularies/:string_key/terms/:uri`
+
+  URI should be encoded.
```
Comment (elohanlon) [inline README.md]:
```suggestion
  URI should be URL-encoded.
```

File: README.md (hunk 2 above)
Comment (elohanlon) [reply README.md]: you're right, it should

File: README.md (hunk 12 above)
Comment (cgalarza) [reply README.md]: gotcha

File: README.md [hunk 18]
```diff
@@ -1,24 +1,343 @@
...
+    "total records": 55,
+    "terms": [
+      {
+        "pref_label": 'Carla Galarza',
+        "uri":"https://id.library.columbia.edu/term/1111-2222-3333-...",
+        "authority": "local",
+        "custom_field_1": "something"
+      },
+      {
+        "pref_label": 'Eric O'Hanlon',
```
Comment (elohanlon) [inline README.md]:
```suggestion
        "pref_label": "Eric O'Hanlon",
```

File: README.md [hunk 19]
```diff
@@ -1,24 +1,343 @@
...
+
+  _Body:_
+  ```json
+  {
+    "page": 1,
+    "per_page": 25,
+    "total records": 55,
+    "terms": [
+      {
+        "pref_label": 'Carla Galarza',
```
Comment (elohanlon) [inline README.md]:
```suggestion
        "pref_label": "Carla Galarza",
```

Comment (elohanlon) [reply README.md]: Github was highlighting this value in red, so maybe it prefers double quotes.

File: README.md (hunk 18 above)
Comment (elohanlon) [reply README.md]: Github was also highlighting this value in red, so maybe it prefers double quotes.  Plus, double quotes around the name string are better for the apostrophe in my name, haha.

**Review by elohanlon — APPROVED** [review]

**Pull request merged**